src/infra/flow-system/postgres/src/flow_system_event_store_postgres.rs

```diff
@@ -0,0 +1,257 @@
+                    .await
+            }
+            FlowKey::System(fk_system) => {
+                self.get_system_events(fk_system, maybe_from_id, maybe_to_id)
+                    .await
+            }
+        }
+    }
+
+    async fn save_events(
+        &self,
+        flow_key: &FlowKey,
+        events: Vec<FlowConfigurationEvent>,
+    ) -> Result<EventID, SaveEventsError> {
+        if events.is_empty() {
+            return Err(SaveEventsError::NothingToSave);
+        }
+
+        let mut tr = self.transaction.lock().await;
+        let connection_mut = tr.connection_mut().await?;
+
+        let mut query_builder = match flow_key {
+            FlowKey::Dataset(fk_dataset) => {
+                let mut query_builder = QueryBuilder::<Postgres>::new(
+                    r#"
+INSERT INTO dataset_flow_configuration_events (dataset_id, dataset_flow_type, event_type, event_time, event_payload)
+"#,
+                );
+
+                query_builder.push_values(events, |mut b, event| {
+                    b.push_bind(fk_dataset.dataset_id.to_string());
+                    b.push_bind(fk_dataset.flow_type);
+                    b.push_bind(event.typename());
+                    b.push_bind(event.event_time());
+                    b.push_bind(serde_json::to_value(event).unwrap());
+                });
+
+                query_builder
+            }
+            FlowKey::System(fk_system) => {
+                let mut query_builder = QueryBuilder::<Postgres>::new(
+                    r#"
+INSERT INTO system_flow_configuration_events (system_flow_type, event_type, event_time, event_payload)
+"#,
+                );
+
+                query_builder.push_values(events, |mut b, event| {
+                    b.push_bind(fk_system.flow_type);
+                    b.push_bind(event.typename());
+                    b.push_bind(event.event_time());
+                    b.push_bind(serde_json::to_value(event).unwrap());
+                });
+
+                query_builder
+            }
+        };
+
+        query_builder.push("RETURNING event_id");
+
+        #[derive(FromRow)]
+        struct ResultRow {
+            event_id: i64,
+        }
+
+        let rows = query_builder
+            .build_query_as::<ResultRow>()
+            .fetch_all(connection_mut)
+            .await
+            .int_err()?;
+        let last_event_id = rows.last().unwrap().event_id;
+
+        Ok(EventID::new(last_event_id))
+    }
+
+    async fn len(&self) -> Result<usize, InternalError> {
+        let mut tr = self.transaction.lock().await;
+        let connection_mut = tr.connection_mut().await?;
+
+        let result = sqlx::query!(
+            r#"
+SELECT MAX(event_count) as count
+FROM (SELECT COUNT(event_id) as event_count
+      FROM dataset_flow_configuration_events
+      UNION ALL
+      SELECT COUNT(event_id) as event_count
+      FROM dataset_flow_configuration_events) as counts;
+"#,
```

Comment:
I would expect to see a sum of `COUNT(event_id) from 2 tables.

Reply:
Fixed